nx-X11/programs/Xserver/hw/nxagent/Reconnect.c

```diff
@@ -438,22 +438,10 @@ Bool nxagentReconnectSession(void)
 
   if (nxagentKeyboard != NULL)
   {
-    int size;
+    nxagentOldKeyboard = strndup(nxagentKeyboard, strlen(nxagentKeyboard));
+    /* FIXME: check if nxagentOldKeyboard == NULL and abort */
```

Comment:
*shrug* then keep it as-is. In any case, allocation failures are likely to not go away and cascade, if you don't free other data first. An allocation failure need not mean that no more memory is available, it's also possible that an application broke the allocator by writing random data to the stack or the heap. I have no idea what happens on allocation failures (for instance, at what point nxagent dies), never actually tested it.